src/main/scala/org/fortysevendeg/sparksftp/common/HiveUserData.scala

```diff
@@ -39,9 +44,29 @@ object HiveUserData {
     salariesDataFromHive.show(false)
 
     //Excerpt from the joined table
-    sparkSession.sql("select name,salary from user_salary").show(false)
+    val userSalaries = sparkSession.sql("select name,salary from user_salary")
+    userSalaries.show(false)
+
+    HiveUserData(userDataFromHive, salariesDataFromHive, userSalaries)
+  }
+
+  def persistAndReadUserData(
+      sparkSession: SparkSession,
+      users: DataFrame,
+      salaries: DataFrame
+  ): IO[HiveUserData] = IO {
+    HiveUserData.persistUserData(sparkSession, users, salaries)
+    HiveUserData.readUserData(sparkSession)
```

Comment:
Each of these functions should return `IO`

Reply:
Changed in: a78818e